### DISCUSSION TITLE
Discord Rich Presence in web app

### DISCUSSION THREAD

**Discussion by JeffersonBledsoe** [music-assistant/support#2041] [Feature requests and ideas]
Not exactly an important feature, but would be nice to have what we're currently listening to appear in Discord like you can do with Spotify and various other music services.

https://discord.com/developers/docs/rich-presence/how-to

**Comment by marcelveldt** [comment]
That feature already exists on the desktop app

**Comment by rophel** [comment]
Desktop app is basically defunct now, so looking for a solution to this.  Could be a browser extension or via an existing extension like PreMiD.

https://docs.premid.app/

**Comment by kepstin** [comment]
Unfortunately, Discord doesn't have a public HTTP api that could be called directly from the Music Assistant server to do rich presence.

It's theoretically possible that this could be implemented server-side using the [Discord Social SDK](https://discord.com/developers/docs/discord-social-sdk/overview) which permits OAuth authentication. Downside is that this is a closed-source library. While it is available for Linux, it's only available for x86-64 and links to glibc and a few other libraries like libasound (alsa), libpulse, libX11.

Also, anyone who wants to use it would have to create their own Discord developer account and application, which requires providing some additional personal information to Discord.

(For what it's worth, I suspect that PreMiD works by having the extension talk to servers they host, and then they use the Discord Social SDK server-side.)
